scm/driver/stash/stash.go

```diff
@@ -74,10 +75,37 @@ func (c *wrapper) do(ctx context.Context, method, path string, in, out interface
 	// if we are posting or putting data, we need to
 	// write it to the body of the request.
 	if in != nil {
-		buf := new(bytes.Buffer)
-		json.NewEncoder(buf).Encode(in)
-		req.Header["Content-Type"] = []string{"application/json"}
-		req.Body = buf
+		switch content := in.(type) {
+		case *contentCreateUpdate:
+			// add the content to the multipart
+			var b bytes.Buffer
+			w := multipart.NewWriter(&b)
+			// add the other fields
+			if content.Message != "" {
+				_ = w.WriteField("content", string(content.Content))
```

Comment:
please handle these errors. Aware they're repetitive but could be more palatable with something like:

```go
type multipartWriter struct {
    w mime. Writer
    err error
}

func (multipartWriter *mw) Write(f, v string) {
    if mw.err == nil {
        return
    }
    if v == "" {
        return
    }
    _, err = mw.WriteField(f, v)    
}
```

```go
case *contentCreateUpdate:
    var b bytes.Buffer
    mw := multipartWriter{w: multipart.NewWriter(&b)}
    mw.write("content", string(content.Content))
    mw.write("message", content.Message)
    mw.write("branch", content.Branch)
    mw.write("sha", content.Sha)
    if mw.err != nil {
        return nil, fmt.Errorf("error writing multipart-content. err: %s", mw.err)
    }
```
see https://go.dev/blog/errors-are-values for background

Reply:
Did this change in another PR